slsim/lens.py

```diff
@@ -320,27 +277,36 @@ def deflector_redshift(self):
     def source_redshift(self):
         """
 
-        :return: source redshift
+        :return: list of source redshifts
         """
-        return self.source.redshift
+        source_redshift_list = []
+        for source in self.source:
+            source_redshift_list.append(source.redshift)
+        return source_redshift_list
 
     @property
     def external_convergence(self):
         """
 
         :return: external convergence
         """
-        _, _, kappa_ext = self.los_linear_distortions
-        return kappa_ext
+        kappa_list = []
+        for los_linear_dist in self.los_linear_distortions:
+            _, _, kappa_ext = los_linear_dist
+            kappa_list.append(kappa_ext)
+        return kappa_list
 
     @property
     def external_shear(self):
         """
 
         :return: the absolute external shear
         """
-        gamma1, gamma2, _ = self.los_linear_distortions
-        return (gamma1**2 + gamma2**2) ** 0.5
+        shear_list = []
+        for los_linear_dist in self.los_linear_distortions:
```

Comment:
I am not sure I understand the looping through here. There is only one line of sight per lens. The only thing that changes is the lensing efficiency part

Reply:
Yes, it is not necessary to loop through all source. They result in the same value. I was confused by the input of the function!